File: Jack.Presentation/Controllers/API/FamiliaController.vb
Imports System.Net
Imports System.Web.Http

Namespace Controllers.API
    Public Class FamiliaController
        Inherits ApiController

        <HttpGet>
        Public Function GetValues() As IList(Of Model.Familia)

            Dim temp As Model.Familia = Nothing
            Dim lstRetorno As List(Of Model.Familia) = Nothing
            lstRetorno = New List(Of Model.Familia)

            temp = New Model.Familia()
            temp.Codigo = 1
            temp.IsConsistente = "S"
            temp.Contato = "99934-4533"
            temp.DataAtualizacao = DateAndTime.Now
            temp.Familia = "Nome Qualquer1"
            temp.IsSacolinha = "N"
            temp.Status = 1
            temp.Nivel = 1

            lstRetorno.Add(temp)

            temp = New Model.Familia()
            temp.Codigo = 2
            temp.IsConsistente = "S"
            temp.Contato = "2232-3432"
            temp.DataAtualizacao = DateAndTime.Now
            temp.Familia = "Nome Qualquer 2"
            temp.IsSacolinha = "N"
            temp.Status = 1
            temp.Nivel = 99
            lstRetorno.Add(temp)

            temp = New Model.Familia()
            temp.Codigo = 3
            temp.IsConsistente = "S"
            temp.Contato = "93321-4367"
            temp.DataAtualizacao = DateAndTime.Now
            temp.Familia = "Nome Qualquer 3"
            temp.IsSacolinha = "S"
            temp.Status = 2
            temp.Nivel = 2
            lstRetorno.Add(temp)
            Return lstRetorno

        End Function

        <HttpPost>
        Public Sub Salvar(<FromUri> oFamily As Model.Familia)

            'atualizando datas
            oFamily.DataAtualizacao = DateTime.Now()

        End Sub

        <HttpDelete>
        Public Sub Delete(intFamilia As Integer)

        End Sub

    End Class
End Namespace

File: Jack.Presentation.Web/Controllers/API/FamiliaController.cs

using System;
using System.Collections.Generic;
using System.Web.Http;
using Business = Jack.Business;
using Model = Jack.Model;

namespace Controllers.API
{
	public class FamiliaController : ApiController
	{

		[HttpGet()]
		public IList<Model.Familia> GetValues()
		{

			IList<Model.Familia> lstRetorno = null;
			Business.Familia oBusiness = default(Business.Familia);

            try
            {
                oBusiness = new Business.Familia();
                lstRetorno = oBusiness.LoadAll();

            }
            catch (Exception ex)
            {
                lstRetorno = null;
            }
            finally
            {
                oBusiness = null;
            }

			return lstRetorno;

		}

		[HttpGet()]
		public Model.Familia GetValue([FromUri()] int ID)
		{

			Model.Familia oRetorno = default(Model.Familia);
			Business.Familia oBusiness = default(Business.Familia);

            try
            {
                oBusiness = new Business.Familia();
                oRetorno = oBusiness.Find(ID);

            }
            catch (Exception ex)
            {
                oRetorno = null;
            }
            finally
            {
                oBusiness = null;
            }

			return oRetorno;

		}

		[HttpPost()]
		public void Salvar([FromUri()] Model.Familia oFamily)
		{
			//atualizando datas
			oFamily.DataAtualizacao = DateTime.Now;

			Business.Familia oBusiness = default(Business.Familia);
            try
            {
                oBusiness = new Business.Familia();
                oBusiness.Update(oFamily);
            }
            finally
            {
                oBusiness = null;
            }

		}

        [HttpDelete()]
        public void Delete([FromUri()] int ID)
        {
            Business.Familia oBusiness = default(Business.Familia);
            Model.Familia oDelete = default(Model.Familia);
            try
            {
                oBusiness = new Business.Familia();
                oDelete = new Model.Familia();
                oDelete.Codigo = ID;
                oBusiness.Delete(oDelete);

            }
            finally
            {
                oBusiness = null;
            }

        }

	}
}
